source/common/filter/tcp_proxy.cc

```diff
@@ -217,29 +230,38 @@ Network::FilterStatus TcpProxy::initializeUpstreamConnection() {
       config_->stats().downstream_cx_no_route_.inc();
     }
     request_info_.setResponseFlag(AccessLog::ResponseFlag::NoRouteFound);
-    onInitFailure();
+    onInitFailure(Http::Code::ServiceUnavailable);
```

Comment:
nit: It's a little odd to me that we are using HTTP codes here. (I wonder how this header is even being included). Would it be better to have an enum dedicated to this function with members NoUpstream and Timeout or something like that? Then the WS code can convert those into HTTP codes?

Reply:
I thought about doing that, and went with the simpler option for now, but I can do this; I agree it would be nicer.  OTOH, one of my next changes is to send things to the outlier detector from tcp_proxy, and that also uses http codes as it's input.

Reply:
I kind of wonder if we should fix the outlier interfaces for this, since we do the same hack in Redis also.

Reply:
We could just add to its interface a way to directly say "connect failed" and "connect timed out", and internally it can just convert those to the http codes it currently uses.

Reply:
Yeah agreed. If you do this, can you cleanup Redis also?

Reply:
Yes, I'll do that.  Hopefully I'll have a PR the week after thanksgiving.